packages/ui/src/components/organisms/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,196 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useMemo, useEffect, useRef, useCallback, useState} from 'react';
+import {StyleSheet, View, Animated} from 'react-native';
+import {ThemeColors, useThemeColor, Color} from '../../../theme';
+import {Text} from '../../atoms';
+
+type ProgressBarProps = {
+  value?: number;
+  total?: number;
+  showPercent?: boolean;
+  colorRepartition?: {[key: number]: Color};
+  height?: number;
+  style?: object;
+  styleTxt?: object;
+  stripe?: boolean;
+  stripeDuration?: number;
+  stripeWidth?: number;
+};
+
+const ProgressBar = ({
+  value = 0,
+  total = 100,
+  showPercent = true,
+  colorRepartition = {},
+  stripe = true,
+  height = 30,
+  stripeDuration = 1000,
+  stripeWidth = 40,
+  style,
+  styleTxt,
+}: ProgressBarProps) => {
+  const Colors = useThemeColor();
+  const animatedStripe = useRef(new Animated.Value(0)).current;
+  const animatedWidth = useRef(new Animated.Value(0)).current;
+  const [progressBarWidth, setProgressBarWidth] = useState(0);
+
+  const percent = useMemo(
+    () => (total !== 0 ? (value / total) * 100 : 0),
+    [total, value],
+  );
+
+  const progressColors = useMemo(() => {
+    if (Object.keys(colorRepartition).length === 0) {
+      return {
+        0: Colors.errorColor,
+        25: Colors.cautionColor,
+        50: Colors.progressColor,
+        75: Colors.priorityColor,
+        100: Colors.successColor,
+      };
+    }
+    return colorRepartition;
+  }, [Colors, colorRepartition]);
+
+  let color: Color = Colors.cautionColor;
+  Object.keys(progressColors).forEach(key => {
+    const threshold = parseInt(key, 10);
+    if (percent >= threshold && progressColors[threshold]) {
+      color = progressColors[threshold];
+    }
+  });
+
+  const styles = useMemo(
+    () => getStyles(Colors, height, color, percent, stripeWidth),
+    [Colors, height, color, percent, stripeWidth],
+  );
+
+  useEffect(() => {
+    Animated.timing(animatedWidth, {
+      toValue: percent,
+      duration: 500,
+      useNativeDriver: false,
+    }).start();
+  }, [percent, animatedWidth]);
+
+  useEffect(() => {
+    const stripeAnimation = Animated.loop(
+      Animated.timing(animatedStripe, {
+        toValue: 1,
+        duration: stripeDuration,
+        useNativeDriver: true,
+      }),
+    );
+    stripeAnimation.start();
+  }, [animatedStripe, stripeDuration, value]);
+
+  const translateX = animatedStripe.interpolate({
+    inputRange: [0, 1],
+    outputRange: [-stripeWidth, progressBarWidth],
+  });
+
+  const renderPercent = useCallback(() => {
+    if (showPercent) {
+      return (
+        <Text style={[styles.text, styleTxt]}>{`${percent.toFixed(2)}%`}</Text>
+      );
+    } else {
+      return <Text style={[styles.text, styleTxt]}>{`${value}/${total}`}</Text>;
+    }
+  }, [percent, showPercent, styles, styleTxt, total, value]);
```

Comment:
Still some issue on text display, when outside of progress bar it should be on the close to progress border as a mirror effect of the normal layout

And the text is still cut on some value, if think disposition should be based on text length AND progress width instead of percent value

Reply:
Maybe we can add a props to know if value should always be centered in container or if it should follow progress bar (idea from Marc :innocent: )

Also, progress bar is not centered anymore in container